### PR TITLE
Implement Application Completion Workflow with Certificate Generation and Email Notification

### LINKED ISSUE #122: Add "COMPLETE" state for applications.
**Description:**
A clear and concise description of the implementation.

Implement a new "COMPLETE" state for mentee applications, allowing admins to update the application status once the program has concluded.

**Tasks:**
Add the tasks need to complete to achieve the implementation.

- Introduce a new "COMPLETE" status in the ApplicationStatus enum.
- Restrict the visibility and functionality of the "COMPLETE" button to admin users only.
- Implement functionality to send an email notification to the mentee when their application status is updated to "COMPLETE".
- Develop functionality to generate a completion certificate for the mentee.
- Ensure the certificate includes relevant details such as the mentee's name, program name, completion date, and any other pertinent information.
- Automatically attach the generated certificate to the email sent to the mentee.

**Acceptance Criteria:**
Implementation should be able to perform these functionalities. 

- Admins can change the application status to "COMPLETE" once the program is over.
- An email is sent to the mentee when their application status is updated to "COMPLETE".
- The email includes an attached certificate of completion for the mentee.

**Additional Information:**
Add additional information 

**Related Dependencies or References:**
Add related dependencies and references relating to the implementation.

- Consult documentation or libraries for generating PDF certificates programmatically.

### PR DESCRIPTION
**Pull request by mayura-andrew**
## Purpose

The purpose of this PR is to fix #122

## Goals

- Enable admins to change the application status to 'COMPLETE'.
- Notify mentees via email upon status change.
- Generate a completion certificate for each mentee and attach it to the notification email.
## Approach

To achieve the above goals, the following steps were taken:

- Integrated the pdf-lib library (version 1.17.1) to modify and generate completion certificates in PDF format.
- Updated utils.ts to support sending emails with PDF attachments.
- Created a separate function dedicated to handling the PDF generation and email notification logic
- Created a new folder to store certificates

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

## Test environment
 
- OS: Linux Mint 21.3
- Browser Firefox
- Version 128.0
- Nodejs 20.15.0

## Learning

During the development of this feature, I explored several resources to understand how to effectively use the pdf-lib library for PDF manipulation and how to attach files to emails programmatically. This included reviewing documentation, blog posts, and tutorials related to pdf-lib and email handling in Node.js applications.

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -145,6 +153,28 @@ export const getEmailContent = (
           We do offer the possibility for you to apply again next time if you meet the eligibility criteria. We invite you to stay engaged with us by attending our events, reaching out to our admissions team, and taking advantage of any opportunities to connect with our current students and alumni.<br /><br />
           Thank you again for considering our program and for the time you invested in your application. We wish you all the best in your future endeavours.`
         }
+      case ApplicationStatus.COMPLETED: {
+        const pdfFileName = await generateCertificate(
+          name,
+          './src/certificates/certificate_template.pdf',
+          `./src/certificates/${name}_certificate.pdf`
```
Comment (anjula-sack) [inline src/utils.ts:160]: Shall we use a unique id here? There might be a chance of having the same name. Also you need to update the certificate_id field of the mentee with file path as well.

 
```suggestion
          `./src/certificates/${name}_certificate.pdf`
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/utils.ts (hunk 1 above)
Comment (mayura-andrew) [reply src/utils.ts:160]: okay sure

File: src/enums/index.ts [hunk 2]
```diff
@@ -12,7 +12,8 @@ export enum EmailStatusTypes {
 export enum ApplicationStatus {
```
Comment (anjula-sack) [inline src/enums/index.ts:12]: Shall we rename this to `MenteeStatus` and create separate one form Mentors? Cuz mentors don't need a completed state. 

```suggestion
export enum MenteeStatus {
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/utils.ts (hunk 1 above)
Comment (mayura-andrew) [reply src/utils.ts:160]: image
here I used uuid library to generate unique values

**Review by anjula-sack — COMMENTED** [review]
Can you resolve the conflicts? @mayura-andrew

**Comment by mayura-andrew** [comment]
> Can you resolve the conflicts? @mayura-andrew
I'm working on this

File: package.json [hunk 3]
```diff
@@ -35,11 +35,13 @@
...
     "passport-google-oauth20": "^2.0.0",
     "passport-jwt": "^4.0.1",
+    "pdf-lib": "^1.17.1",
     "passport-linkedin-oauth2": "github:auth0/passport-linkedin-oauth2#v3.0.0",
     "pg": "^8.10.0",
     "reflect-metadata": "^0.1.13",
     "ts-node": "^10.9.1",
-    "typeorm": "^0.3.16"
+    "typeorm": "^0.3.16",
+    "uuid": "^10.0.0"
```
Comment (anjula-sack) [inline package.json:44]: https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/enums/index.ts (hunk 2 above)
Comment (mayura-andrew) [reply src/enums/index.ts:12]: For now, could we avoid changing this? There is a lot of refactoring involved.

**Review by anjula-sack — APPROVED** [review]
Thank you! @mayura-andrew

**Pull request merged**